### PR TITLE
We now have a Kotlin program to blink a LED on stm32f4_disco.

### PR DESCRIPTION
**Pull request by alexander-gorshenev**
This pull request depends on #1343 and #1336 to provide required functionality.

### CODE REVIEW

File: samples/zephyr/build.sh [hunk 1]
```diff
@@ -22,14 +22,35 @@ fi
...
+
+. $DIR/c_interop/platforms/$BOARD.sh
+
+# We need generated headers to be consumed by `cinterop`, 
+# so we preconfigure the project with `make zephyr`.
+# But `program.o` should exist at that stage to be detected by the configuration tools.
+# So we create an empty `program.o` above, and then override it with the compiler result below.
+# TODO: do something with this mess.
+
+rm -f program.o
```
Comment (olonho) [inline samples/zephyr/build.sh]: Looks ugly indeed.

File: samples/zephyr/c_interop/platforms/stm32f4_disco.def [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Keep this one board independent.
+
+headers = autoconf.h zephyr.h board.h device.h gpio.h
+compilerOpts = -DKERNEL -DUSE_FULL_LL_DRIVER -DUSE_HAL_DRIVER -D_FORTIFY_SOURCE=2 -D__ZEPHYR__=1 -ffreestanding -std=c99 -nostdinc
+
+---
+
+// cinterop doesn't support `#define FOO "string"` constructs as of yet.
```
Comment (olonho) [inline samples/zephyr/c_interop/platforms/stm32f4_disco.def]: Slava’s commit will fix it, maybe just wait a bit?

File: samples/zephyr/c_interop/platforms/stm32f4_disco.sh [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+fi
+
+# TODO: we need a robust automated way to ask Zephyr for 
+# all these $ZEPHYR_BASE based paths and the proper defines.
+# TODO: The D flag in `-copt -Dxxx` sequence is interpreted to be a jvm property,
+# so we workaround the using -Xclang.
+
+cinterop -def $DIR/c_interop/platforms/$BOARD.def \
+        -pkg platform.zephyr.$BOARD \
+        -copt '-Xclang -DSTM32F407xx' \
```
Comment (olonho) [inline samples/zephyr/c_interop/platforms/stm32f4_disco.sh]: Why not use multientry -compilerOpts?

File: samples/zephyr/build.sh [hunk 4]
```diff
@@ -22,14 +22,30 @@ fi
...
+
+mkdir -p $DIR/build/kotlin
+
+konanc $DIR/src/main.kt \
+        -target zephyr_$BOARD \
+        -r $DIR/c_interop/platforms/build \
+        -l $BOARD \
+        -opt -g -o $DIR/build/kotlin/program || exit 1
```
Comment (olonho) [inline samples/zephyr/build.sh]: -g?

Comment (alexander-gorshenev) [reply samples/zephyr/build.sh]: yep, to be useful as an interesting sample.

**Pull request merged**